core/src/main/java/org/apache/iceberg/IncrementalDataTableScan.java

```diff
@@ -34,7 +35,7 @@
 class IncrementalDataTableScan extends DataTableScan {
 
   IncrementalDataTableScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
-    super(ops, table, schema, context.useSnapshotId(null));
+    super(ops, table, schema, context.useSnapshotId(null), new LoggingScanReporter());
```

Comment:
Rather than changing so many method signatures, why not add `ScanReporter` to the `TableScanContext`?

Reply:
I haven't considered that, but that makes perfect sense. Updated the code